test_conformance/images/kernel_read_write/test_write_1D.cpp

```diff
@@ -215,6 +215,13 @@ int test_write_image_1D( cl_device_id device, cl_context context, cl_command_que
         }
         else // Either CL_MEM_ALLOC_HOST_PTR, CL_MEM_COPY_HOST_PTR or none
         {
+            char *host_ptr = NULL;
+
+            if (gMemFlagsToUse & CL_MEM_COPY_HOST_PTR)
+            {
+                host_ptr = imageValues;
+            }
```

Comment:
Would it be more robust to leave all of the calls into the image helpers unchanged and to instead move logic similar to this into the image helpers themselves?

Reply:
Agree, but this may apply to other cl_mem_flags as well. Can we do this in a separate PR?

Reply:
Would you mind taking a quick look at this vs. pushing off to a separate PR?  I think it is a much more robust solution and it'd make many parts of this PR unnecessary.

Reply:
Specifically, to get equivalent behavior as this PR I think we'd only need to check for `CL_MEM_USE_HOST_PTR` and `CL_MEM_COPY_HOST_PTR`, to ensure we only pass a non-NULL host pointer when we are allowed to:

> CL_INVALID_HOST_PTR if host_ptr is NULL and CL_MEM_USE_HOST_PTR or CL_MEM_COPY_HOST_PTR are set in flags or if host_ptr is not NULL but CL_MEM_COPY_HOST_PTR or CL_MEM_USE_HOST_PTR are not set in flags.

We wouldn't need to check for other `cl_mem_flags`.

Reply:
Updated the change with checks moved to ImageHelper and earlier changes reverted from test_conformance/image/.
Only a small note - If we decide to add new tests to check for NO_HOST_PTR behavior to ensure if implementations return expected error or not, we may have to revisit this or use a different path to avoid overriding NULL host-ptr.